tests/test_namespace.py

```diff
@@ -0,0 +1,25 @@
+import os
+import pytest
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "html_file, expected_xpath",
+    [
+        (
+            "mock_html/namespace.html",
+            '//html/body/*[name()="sc:visitoridentification"]/div',
+        ),
+    ],
+)
+async def test_xpath(tarsier, async_page, html_file, expected_xpath):
+    html_file_path = os.path.abspath(os.path.join(os.path.dirname(__file__), html_file))
+    await async_page.goto(f"file://{html_file_path}")
+
+    page_text, tag_to_xpath = await tarsier.page_to_text(
+        async_page, tag_text_elements=True
+    )
+    assert tag_to_xpath[0] == expected_xpath, (
+        f"tag_to_xpath does not match expected xpath for "
+        f"{html_file}. Got: {tag_to_xpath}"
+    )
```

Comment:
doesn't need to be parameterized given its a single test. It also kind of makes things hard to understand since everything is named so generally instead of name space specific
```suggestion
            ,
            ,
        ),
    ],
)
async def test_xpath(tarsier, async_page, html_file, expected_xpath):
    html_with_namespace_path = os.path.abspath(os.path.join(os.path.dirname(__file__), "mock_html/namespace.html"))
    await async_page.goto(f"file://{html_with_namespace_path}")

    _, tag_to_xpath = await tarsier.page_to_text(
        async_page, tag_text_elements=True
    )
    assert len(tag_to_xpath) == 1, "The page contains only a single tag"
    assert tag_to_xpath[0] == '//html/body/*[name()="sc:visitoridentification"]/div', (
        f"Namespaces within the xpath were not correctly handled "
        f"Got: {tag_to_xpath}"
    )
```